drivers/include/mlx90393.h

```diff
@@ -0,0 +1,403 @@
+} mlx90393_resolution_t;
+
+/**
+ * @brief   Oversampling ratio of the MLX90393 sensor
+ */
+typedef enum {
+    MLX90393_OSR_0,                 /**< Oversampling ratio 0 */
+    MLX90393_OSR_1,                 /**< Oversampling ratio 1 */
+    MLX90393_OSR_2,                 /**< Oversampling ratio 2 */
+    MLX90393_OSR_3,                 /**< Oversampling ratio 3 */
+} mlx90393_oversampling_ratio_t;
+
+/**
+ * @brief   Digital filter applicable to ADC of the MLX90393 sensor
+ */
+typedef enum {
+    MLX90393_DIG_FILT_0,            /**< Digital filter 0 */
+    MLX90393_DIG_FILT_1,            /**< Digital filter 1 */
+    MLX90393_DIG_FILT_2,            /**< Digital filter 2 */
+    MLX90393_DIG_FILT_3,            /**< Digital filter 3 */
+    MLX90393_DIG_FILT_4,            /**< Digital filter 4 */
+    MLX90393_DIG_FILT_5,            /**< Digital filter 5 */
+    MLX90393_DIG_FILT_6,            /**< Digital filter 6 */
+    MLX90393_DIG_FILT_7,            /**< Digital filter 7 */
+} mlx90393_digital_filter_t;
+
+/**
+ * @brief   Oversampling rates of magnetic and temperature sensors
+ */
+typedef struct {
+    mlx90393_oversampling_ratio_t mag;      /**< Oversampling ratio magnetic sensor */
+    mlx90393_oversampling_ratio_t temp;     /**< Oversampling ratio temperature sensor */
+} mlx90393_oversampling_t;
+
+/**
+ * @brief   Tresholds for wake-up on change mode
+ */
+typedef struct {
+    int xy;                             /**< Treshold for x and y axes */
+    int z;                              /**< Tresold for z axis */
+    int temp;                           /**< Tresold for temperature */
+} mlx90393_treshold_t;
+
+/**
+ * @brief   Device configuration parameters
+ */
+typedef struct {
+#if MODULE_MLX90393_SPI || DOXYGEN
+    spi_t spi;                          /**< SPI bus */
+    gpio_t cs_pin;                      /**< Connected chip select pin */
+    spi_clk_t clk;                      /**< clock speed for the SPI bus */
+#elif MODULE_MLX90393_I2C || DOXYGEN
+    i2c_t i2c;                          /**< I2C device */
+    uint8_t addr;                       /**< Magnometer I2C address */
+#endif
+    mlx90393_mode_t mode;                   /**< Measurement mode */
+    gpio_t int_pin;                         /**< Interrupt pin */
+    mlx90393_gain_t gain;                   /**< Analog chain gain */
+    mlx90393_resolution_t resolution;       /**< Desired 16-bit output value from the 19-bit ADC */
+    mlx90393_odr_t odr;                     /**< Output data rate */
+    mlx90393_oversampling_t oversampling;   /**< Oversampling ratio */
+    mlx90393_digital_filter_t dig_filt;     /**< Digital filter applicable to ADC */
+    mlx90393_treshold_t treshold;           /**< Treshold for wake-up on change mode */
+} mlx90393_params_t;
+
+/**
+ * @brief   Device descriptor for the MLX90393 magnetometer
+ */
+typedef struct {
+    mlx90393_params_t params;           /**< Device configuration parameters */
```

Comment:
```suggestion
    mlx90393_params_t *params;          /**< Device configuration parameters */
```
Es gibt keinen Grund die konstanten Parameter zu kopieren, ein Pointer reicht. Da die `static const` Parameter im ROM abgelegt werden, das `mlx90393_t` Gerät jedoch im RAM gehalten wird, spart dies etliches an RAM.